instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/DefaultSpanStatusExtractor.java

```diff
@@ -15,7 +16,7 @@
 
   @Override
   public StatusCode extract(
-      REQUEST request, @Nullable RESPONSE response, @Nullable Throwable error) {
+      REQUEST request, @Nullable RESPONSE response, SpanKind kind, @Nullable Throwable error) {
```

Comment:
Instead of introducing a `kind` param, how about deciding the behavior basing on the attributes extractor used? I.e. you could check `extractor instanceof HttpServerAttributesExtractor` and choose the behavior based on that

Reply:
+1 to choosing right implementation ASAP. Although it does not make any difference performance-wise.

Reply:
But, that would mean that no one could build a kind-oriented attributes extractor (independent of this particular one for http server spans), which still seems like a limitation that might not be desirable.

Reply:
I've never met an `instanceof` that I didn't despise (and later regret).  😁

Reply:
would splitting into `HttpClientSpanStatusExtractor` and `HttpServerSpanStatusExtractor` work?

we've been finding it useful to split out other extractors based on client/server, e.g. `Http(Client|Server)AttributesExtractor` and `Net(Client|Server)AttributesExtractor` (though there's an open discussion on the latter in #4408)

Reply:
I considered it but was too concerned it might cause problems with frameworks (and consequently instrumentations) that aren't purely client/server, right? Isn't there instrumentation that can generate both SERVER and CLIENT spans? I'm thinking about messaging systems and http/2 server push, for example.

This was also the thinking behind the "late binding", if you will (I think this is what @iNikem was getting at with the "ASAP" comment).

If this isn't true and we don't think it'll be a problem, we could try that out.

Reply:
> I considered it but was too concerned it might cause problems with frameworks (and consequently instrumentations) that aren't purely client/server, right? Isn't there instrumentation that can generate both SERVER and CLIENT spans? I'm thinking about messaging systems and http/2 server push, for example.

They already will have to provide the correct `SpanKindExtractor`. Which means that they will most probably have separate instrumenters per kind. Thus separate `SpanStatusExtractor` will not be a problem either.

Reply:
> Isn't there instrumentation that can generate both SERVER and CLIENT spans?

I think all of these are creating two instrumenters already, one to handle server spans and one to handle client spans.